crates/ruma-identifiers-validation/src/room_id.rs

```diff
@@ -1,5 +1,10 @@
-use crate::{validate_delimited_id, Error};
+use crate::{server_name, validate_id, Error};
 
 pub fn validate(s: &str) -> Result<(), Error> {
-    validate_delimited_id(s, &['!'])
+    validate_id(s, &['!'])?;
+    if let Some(colon_idx) = s.find(':') {
```

Comment:
I am wondering if it is right to check this here? What if a future format allows colons?

Reply:
If we don't do it like this, I think we no longer have a chance of reliably detecting server names?

Reply:
I was just thinking of calling `<&ServerName>::try_from(str)` in `server_name()` instead of during validation.

Reply:
Yeah but then the return type becomes more complicated. Should it be `Result<Option<_>, _>`? `Option<Result<_, _>>`?

Reply:
I was thinking of just ignoring errors there. Although for debugging purposes maybe it's not the right thing to do.

Reply:
I've now implemented this, with a tracing warning if a colon is found but the string after it is not a valid server name.